File: models/Document.js
const mongoose = require('mongoose');

const documentSchema = new mongoose.Schema({
  visiteurId:    { type: mongoose.Schema.Types.ObjectId, ref: 'Visiteur', default: null },
  nomFichier:    { type: String, required: true, maxlength: 255 },
  cheminFichier: { type: String, required: true, maxlength: 500 },
  typeMime:      { type: String, required: true, maxlength: 50 },
  tailleFichier: { type: Number, required: true },
  estArchive:    { type: Boolean, default: false },
}, { timestamps: true });

module.exports = mongoose.model('Document', documentSchema);

File: models/document.py
from typing import Optional
from datetime import datetime
from beanie import Document, PydanticObjectId
from pydantic import Field


class DocumentScan(Document):
    visiteur_id: Optional[PydanticObjectId] = None

    nom_fichier: str = Field(..., max_length=255)
    chemin_fichier: str = Field(..., max_length=500)
    type_mime: str = Field(..., max_length=50)

    taille_fichier: int

    est_archive: bool = False

    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "documents"